### ISSUE TITLE
Linux: Project Manager may crash after switching enabled gems

### ISSUE THREAD

**Issue by LB-BartoszDudzinski** [o3de/o3de#5129]
**Describe the bug**
Project Manager may crash after a gem is enabled or disabled. Selecting a different template for a project produces the same result. It occurs in Single and Multi configuration builds, however not all of the machines are affected. The issue was reproduced on 2 out of 3 machines.

**Steps to reproduce**
1. Open a terminal in the AutomatedTesting project folder and execute the following commands:
 
`cmake -B build/linux -S . -G "Unix Makefiles" -DCMAKE_C_COMPILER=clang-12 -DCMAKE_CXX_COMPILER=clang++-12 -DCMAKE_BUILD_TYPE=profile`

`cmake --build build/linux --target AutomatedTesting.GameLauncher Editor`

2. Launch the Project Manager.
3. Edit AutomatedTesting settings.
4. Enable or disable any gem.

**Expected behavior**
Gem selection is applied successfully. Project Manager remains stable.

**Actual behavior**
Project Manager crashes.

**Screenshots/Video**

https://user-images.githubusercontent.com/86952009/139437383-aa7a5d27-a936-42a3-84cc-0b04cbe37a80.mp4

**Found in Branch**
Stabilization/2110 (https://github.com/o3de/o3de/commit/9dac7270928d67e1afdba0fe842df0358fb27563)

**Desktop/Device:**
- Device: PC
- OS: Ubuntu
- Version 20.04
- CPU Intel I5-9600k
- GPU RTX 2060
- Memory 16GB

**Comment by monroegm** [comment]
@pollend Have you fixed this already? I can't assign this issue to you. Are you part of the o3de community?

**Comment by pollend** [comment]
Yep I think it's fixed. No I don't think I'm apart of the o3de community.
